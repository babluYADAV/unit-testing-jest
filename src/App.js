import "./App.css";
import CommentForm from "./Components/CommentForm";
import CommentList from "./Components/CommentList";
import Joke from "./Components/Joke";

function App() {
  const comment = [
    { id: 1, comment: "comment one" },
    { id: 2, comment: "comment two" },
  ];
  return (
    <div className="App" data-testid="myrootdiv">
      <CommentForm />
      <CommentList allCommnents={comment} />
      <Joke />
    </div>
  );
}

export default App;
