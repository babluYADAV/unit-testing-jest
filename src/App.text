// import { render, screen, logRoles } from "@testing-library/react";
// import App from "./App";

// // test("renders learn react link", () => {
// //   render(<App />);
// //   const text = screen.queryByTestId("text").innerHTML;
// //   expect(text).toBe(
// //     "React unit testing with jest and react-testing-library! ðŸ˜˜ðŸ’•ðŸ˜‚"
// //   );
// // });

// test("test 1", () => {
//   render(<App />);
//   logRoles(screen.getByTestId("myrootdiv"));
//   const buttonElem = screen.getByRole("button", {
//     name: "test button",
//     exact: false,
//   });
//   expect(buttonElem).toBeInTheDocument();
// });

// import "./App.css";

// function App() {
//   return (
//     <div className="App" data-testid="myrootdiv">
//       <h1>Testing basics</h1>
//       <input type="text" placeholder="type name here" />
//       <button>test button</button>
//       <ul>
//         <li>item one </li>
//         <li>item two </li>
//       </ul>
//     </div>
//   );
// }

// export default App;
