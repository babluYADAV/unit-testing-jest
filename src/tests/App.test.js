import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";

describe("App test", () => {
  test("snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should app render", () => {
    render(<App />);
    const rootComp = screen.getByTestId("myrootdiv");
    expect(rootComp).toBeInTheDocument();
  });
});
