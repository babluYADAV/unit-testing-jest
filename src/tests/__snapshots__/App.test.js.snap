// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test snapshot 1`] = `
<div
  className="App"
  data-testid="myrootdiv"
>
  <div>
    <h2>
      comment form
    </h2>
    <input
      onChange={[Function]}
      placeholder="write your comment here"
      type="text"
      value=""
    />
    <input
      defaultChecked={false}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="checkbox"
    >
      i agree to terms and conditions
    </label>
    <button
      disabled={true}
      onClick={[Function]}
    >
      Comment
    </button>
  </div>
  <div>
    <ul>
      <li>
        comment one
      </li>
      <li>
        comment two
      </li>
    </ul>
  </div>
  <ul />
</div>
`;
